src/toolkits/drawing_classifier/drawing_classifier.cpp

```diff
@@ -25,20 +26,449 @@ using neural_net::model_backend;
 using neural_net::model_spec;
 using neural_net::shared_float_array;
 
+struct result {
+  shared_float_array loss_info;
+  shared_float_array accuracy_info;
+  data_iterator::batch data_info;
+};
+
 }  // namespace
 
-std::unique_ptr<model_spec> drawing_classifier::init_model() const {
+std::unique_ptr<model_spec> drawing_classifier::init_model(
+    bool use_random_init) const {
   std::unique_ptr<model_spec> result(new model_spec);
   return result;
 }
 
+void drawing_classifier::init_options(
+    const std::map<std::string, flexible_type>& opts) {
+  // Define options.
+  options.create_integer_option(
+      "batch_size",
+      "Number of training examples used per training step",
+      256,
+      1,
+      std::numeric_limits<int>::max());
+  options.create_integer_option(
+      "max_iterations",
+      "Maximum number of iterations/epochs made over the data during the"
+      " training phase",
+      500,
+      1,
+      std::numeric_limits<int>::max());
+  options.create_integer_option(
+      "random_seed",
+      "Seed for random weight initialization and sampling during training",
+      FLEX_UNDEFINED,
+      std::numeric_limits<int>::min(),
+      std::numeric_limits<int>::max());
+
+  // Validate user-provided options.
+  options.set_options(opts);
+
+  // Choose a random seed if not set.
+  if (options.value("random_seed") == FLEX_UNDEFINED) {
+    std::random_device random_device;
+    int random_seed = static_cast<int>(random_device());
+    options.set_option("random_seed", random_seed);
+  }
+
+  // Write model fields.
+  add_or_update_state(flexmap_to_varmap(options.current_option_values()));
+}
+
+std::tuple<gl_sframe, gl_sframe> drawing_classifier::init_data(
+      gl_sframe data, variant_type validation_data) const {
+  gl_sframe train_data;
+  gl_sframe val_data;
+  if (variant_is<gl_sframe>(validation_data)) {
+    train_data = data;
+    val_data = variant_get_value<gl_sframe>(validation_data);
+    /* TODO: Re-evaluate the following log_and_throw.
+     *       Enabling it will trigger a unit test failure!
+     */
+    // if (val_data.empty()) {
+    //   log_and_throw("Input SFrame either has no rows or no columns. A "
```

Comment:
TODO is enough and maybe comments can be removed for the same.

Reply:
Done.